vm.c

```diff
@@ -3667,6 +3667,10 @@ Init_VM(void)
     fcore = rb_class_new(rb_cBasicObject);
     rb_set_class_path(fcore, rb_cRubyVM, "FrozenCore");
     RBASIC(fcore)->flags = T_ICLASS;
+    // Classes are allocated pre-tenured, we're overwriting this by manually assigning
+    // flags, so we reset the age in order to make sure the promoted bit and the age
+    // bits are in sync.
+    //rb_reset_age(fcore);
```

Comment:
This can be removed right?

Reply:
I opened #7940 which should solve this issue.

Reply:
Thanks for this @peterzhu2118 - Now that #7940 has been merged I've rebased so this is no longer an issue.